MapboxMobileEvents/Categories/NSUserDefaults+MMEConfiguration.m

```diff
@@ -286,10 +286,10 @@ - (NSString *)mme_userAgentString {
     dispatch_once(&onceToken, ^{
 
         userAgent = [NSString stringWithFormat:@"%@/%@ (%@; v%@)",
-            NSBundle.mme_mainBundle.infoDictionary[(id)kCFBundleNameKey] ?: NSBundle.mme_mainBundle.bundlePath.lastPathComponent.stringByDeletingPathExtension,
```

Comment:
Non-Blocker:
This may be of little value given the life expectancy of this repo.

There's a a number of repetitive calls here where a new piece of information could forget to include the additional stripping. To prevent this, the stripping could take place on the collection rather than individually?

Option: Strip After setting up array
```
// Collection of Input Values
NSArray * values = [
   [NSBundle.mme_mainBundle objectForInfoDictionaryKey:(id)kCFBundleNameKey] ?: NSBundle.mme_mainBundle.bundlePath.lastPathComponent.stringByDeletingPathExtension,
   NSBundle.mme_mainBundle.mme_bundleVersionString,
    NSBundle.mme_mainBundle.bundleIdentifier,
    NSBundle.mme_mainBundle.infoDictionary[(id)kCFBundleVersionKey]
];

// Strip Out invalid characters to form valid arguments
NSMutableArray *arguments = [NSMutableArray arrayWithCapacity:[values count]];
[self enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
        [result addObject:[(NSString*)obj mme_stringByRemovingNonUserAgentTokenCharacters]];
}];
userAgent = [NSString stringWithFormat:@"%@/%@ (%@; v%@)", arguments];
```

Feel free to ignore though, this may see very little value given the differences between the two sections here.

Reply:
Removing comment. this is clutter.